src/qt/sendcoinsdialog.cpp

```diff
@@ -263,6 +263,27 @@ void SendCoinsDialog::on_sendButton_clicked()
     QStringList formatted;
     Q_FOREACH(const SendCoinsRecipient &rcp, currentTransaction.getRecipients())
     {
+        // If user is attempting address reuse, show an informative warning
+        if (model->isUsed(rcp.address.toStdString()))
+        {
+            QString reuseWarningText = tr(
+                        "You have sent to the Bitcoin address %1 already!\n\n"
+                        "Sending to the same address twice is technically not supported and "
```

Comment:
It would not be a bug to stop scanning for transactions paying to an address once it has been used. It "works" only by accident.